background.ts

```diff
@@ -337,14 +338,19 @@ Electron.app.on('window-all-closed', () => {
   Electron.app.dock?.hide();
   // On windows and macOS platforms, we only quit via the notification tray / menu bar.
   // On Linux we close the application since not all distros support tray menu/icons
-  if (os.platform() === 'linux' && !settings.isFirstRun()) {
+  if (os.platform() === 'linux' && !settings.firstRunDialogNeeded()) {
     Electron.app.quit();
   }
 });
 
 Electron.app.on('activate', async() => {
   // On macOS it's common to re-create a window in the app when the
   // dock icon is clicked and there are no other windows open.
+  if (waitForFirstRunDialogCompletion) {
+    console.log('Still processing the first-run dialog: not opening main window');
```

Comment:
It make be a good idea to call `window.restoreWindow` here, in case the user lost it?

Reply:
Not sure what situation you are talking about here; clicking on the dock icon will restore the first-run dialog if it had been hidden behind other windows, or using either the yellow dot, or ⌥-Click on the dock icon.

Reply:
This is getting to be too extreme an edge case. If the user lost the first-run dialog, they can rerun RD and change the default settings.